pyopencl/array.py

```diff
@@ -636,7 +636,7 @@ def _new_with_changes(self, data, offset, shape=None, dtype=None,
         else:
             events = self.events
 
-        return Array(None, shape, dtype, allocator=allocator,
+        return type(self)(None, shape, dtype, allocator=allocator,
```

Comment:
```suggestion
        return self.__class__(None, shape, dtype, allocator=allocator,
```
Would this work, too? This can be a fairly hot code path, and if this helps avoid any function call overhead, it might be worth it. (I don't have data to back this up, but it *feels* faster :)

Reply:
Sure! Looks 👍